### ISSUE TITLE
VCS: Opening File into the first tab doesn't work properly

### ISSUE THREAD

**Issue by uliska** [wbsoft/frescobaldi#1010]
This replaces #1009.

The observations described in #1009 seem wrong. Further investigation leads me to believe the problem is the following:

* When all documents are closed, an empty view is created
  * This view is of course not vcs tracked
  * Consequently no VCSDiffArea is created
* When in that state a document from a Git repository is opened it is opened in the already open view
  * `viewmanager.ViewSpace.showDocument()` is not called.
  * `vcs.manager.VCSManager.slotDocumentURLChanged()` *is* called.
  * In that slot `gitrepo.RepoManager.track_document` is called, but there is no VCSDiffArea - which should have been created when the view was created and there was no tracked document.
* Consequently there will be exceptions when `viewmanager.ViewSpace.disconnectView()` will be called upon closing the view.

I can see two approaches to avoid that situation:

1. create a VCSDiffArea in `slotDocumentUrlChanged()` when the new document is vcs tracked and there isn't a widget already
2. *always* create a DiffArea when creating a view but have it collapsed (width=0) and make it visible in `slotDocumentUrlChanged()`.
3. Don't open the document within the empty view but create a new one altogether. But this probably shouldn't be dealt with in `slotDocumentURLChanged()` but in the view manager instead.

**Comment by uliska** [comment]
A similar issue happens when saving a tracked document as a new document. As the new document is *not* tracked things will go wrong in that case. The VCSDiffArea has to be removed in that case.

I *think* (and will try in a minute) that in `vcs.manager.VCSManager.slotDocumentURLChanged()` we have to invoke `sidebar.ViewSpaceSidebarManager.updateView()`, which should be possible by emitting `viewSpace.viewChanged`. This function is responsible to create or remove the Diff area according to the current view's document's state.

**Comment by uliska** [comment]
What's still going wrong after the previous commit is switching from a tracked to an untracked document within a tab/view instance (which happens by "save as..." of a tracked file).

**Comment by uliska** [comment]
Unfortunately there are two issues mixed in here (by me adding a second topic in a comment).

The original issue of loading a tracked file into the empty initial tab seems fixed (since b437f6998397c04514896c5fb4c98c5693a0612f).

The other part, saving a tracked file under a new name, still causes the following traceback for me:

```
Traceback (most recent call last):
  File "/home/urs/git/lilypond/frescobaldi/frescobaldi_app/sidebar/__init__.py", line 228, in updateView
    self._vcsdiffarea.setTextEdit(view)
  File "/home/urs/git/lilypond/frescobaldi/frescobaldi_app/widgets/vcsdiffarea.py", line 37, in setTextEdit
    self._textedit.vcsDocTracker.diff_updated.disconnect(self.update)
TypeError: disconnect() failed between 'diff_updated' and 'update'
```

**Comment by uliska** [comment]
OK, as I said the *original* issue has been fixed, so I created the new issue #1016 to replace this current one.
